src/anyio/_backends/_asyncio.py

```diff
@@ -808,6 +793,10 @@ def task_done(_task: asyncio.Task) -> None:
             self._tasks.remove(task)
             del _task_states[_task]
 
+            if self._on_completed_fut is not None and not self._tasks:
+                if not self._on_completed_fut.done():
```

Comment:
I'm curious – under what circumstances would this check be negative? That is, if there are no tasks, how could `_on_completed_fut` be set already?

Reply:
Hm, perhaps if the host task (and thus the future) was cancelled?

Reply:
yep, removing this check makes 18 tests fail - I've replaced it with a try/catch - should be faster on 3.11+